packages/cdktf-cli/bin/cmds/ui/components/outputs.tsx

```diff
@@ -13,17 +17,75 @@ function sanitize(value: any) {
   return value;
 }
 
+function Output({ name, value }: { name: string; value: TerraformOutput }) {
+  return (
+    <Box key={name}>
+      <Text>
+        {name} = {value.sensitive ? "<sensitive>" : sanitize(value.value)}
+      </Text>
+    </Box>
+  );
+}
+
+function NestedOutput({
+  name,
+  value,
+  indentationLevel = 0,
+}: {
+  name: string;
+  value: NestedTerraformOutputs;
+  indentationLevel?: number;
+}) {
+  // This is a safe-guard against infinite recursion
+  if (indentationLevel > 500) {
+    return <></>;
+  }
+  if (isTerraformOutput(value)) {
+    return (
+      <Box key={name}>
+        <Output name={name} value={value} />
+      </Box>
+    );
+  }
+
+  return (
+    <Box flexDirection="column" key={name} marginLeft={2} marginBottom={1}>
+      <Text bold>{name}</Text>
+      <Box marginLeft={indentationLevel * 2} flexDirection="column">
+        {Object.entries(value)
+          .sort(([k1], [k2]) => {
+            if (isTerraformOutput(value[k1]) && isTerraformOutput(value[k2])) {
+              return k1.localeCompare(k2);
+            }
+
+            if (isTerraformOutput(value[k1])) {
+              return -1;
+            }
+
+            if (isTerraformOutput(value[k2])) {
+              return 1;
+            }
+            return k1.localeCompare(k2);
+          })
```

Comment:
nit: maybe move this into a comparator function with a very explicit name? would make the code more readable.